### ISSUE TITLE
404 articles are returned in search results

### ISSUE THREAD

**Issue by emilghittasv** [mozilla/sumo#1588]
**Steps to reproduce**
1. Type `new test article23` inside the searchbar.
2. Click on the "New:Test Article23" search result.

**Expected behavior**
If the article exists, the article page is successfully displayed. If the article does not exist (deleted) it is not returned as a search result.

**Actual behavior**
The 404 Page not found is displayed

**Desktop:**
 - OS: Windows 10
 - Browser: Firefox

**Additional context**
* Reproducible in `stage` 
* Article in question https://support.allizom.org/en-US/kb/new-test-article2

**Comment by escattone** [comment]
I've confirmed that all of the articles that 404, like [`new-test-article2`](https://support.allizom.org/en-US/kb/new-test-article2), were created in the stage database prior to the stage migration to Postgres, and of course no longer existed when the test data was loaded into stage just prior to migration. This implies that the Elasticsearch index that existed prior to the Postgres migration on stage was updated but not replaced when the `./manage.py es_reindex` command was run after the new test data was loaded, because those 404 articles were indexed when the old data was used on stage, and of course were never deleted (they simply disappeared). I was forced to do the following to confirm this:

- Checked that the currently-loaded test data does not contain these articles.
- After reverting the stage MySQL instance (but not the currently-active Postgres instance) to the old data (the data just prior to the current test data), I found that all of the articles that 404'ed in the current test data, existed in the old data.
